src/couch_replicator_scheduler.erl

```diff
@@ -279,13 +281,12 @@ start_jobs(Count, State) ->
     [start_job_int(Job, State) || Job <- Runnable3],
     ok.
 
-
-stop_jobs(Count, State) ->
-    Running0 = running_jobs(),
+-spec stop_jobs(non_neg_integer(), boolean(), #state{}) -> non_neg_integer().
+stop_jobs(Count, Continuous, State) ->
+    Running0 = running_jobs(Continuous),
     Running1 = lists:sort(fun oldest_job_first/2, Running0),
```

Comment:
that is, let `running_jobs/0` return all running jobs, then filter out the one-shot replications, then sort, then truncate. I get that it seems inefficient, but it makes for simple and focused functions.